### ISSUE TITLE
add instructions in readme that you may have to install gsl libraries if building locally

### ISSUE THREAD

**Issue by sbillinge** [diffpy/diffpy.pdfgui#184]
...and give commands for macos and ubuntu.  See Slack for info.

**Comment by bobleesj** [comment]
On MacOS (Worked for mine M1)

```
brew install gsl
```

On ubuntu

```
sudo apt-get update
sudo apt-get install libgsl-dev
```

The error message before was:

```
clang++ -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/imac/miniconda3/envs/pdffit2_test/include -arch arm64 -fPIC -O2 -isystem /Users/imac/miniconda3/envs/pdffit2_test/include -arch arm64 -I/private/var/folders/yd/2758vf317zggm0n0mfmk0tpw0000gn/T/pip-install-dr9tinaa/diffpy-pdffit2_9a1c50a0b0e64854a8cff6bf93d0732f -I/Users/imac/miniconda3/envs/pdffit2_test/include/python3.11 -c src/extensions/libpdffit2/pdf.cc -o build/temp.macosx-11.0-arm64-cpython-311/src/extensions/libpdffit2/pdf.o -std=c++11 -Wall -Wno-write-strings -O3 -funroll-loops -ffast-math
      src/extensions/libpdffit2/pdf.cc:29:10: fatal error: 'gsl/gsl_errno.h' file not found
      #include <gsl/gsl_errno.h>
               ^~~~~~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang++' failed with exit code 1
```

**Comment by Tieqiong** [comment]
@sbillinge I'm not entirely sure but isn't gsl and g++ only required if we pip install? If people install from Conda (which we want them to do) then this doesn't seem necessary.

**Comment by sbillinge** [comment]
yes, that's right.  It is maybe good enough to leave it here in the issues and then repost to diffpy-users if anyone asks about it?

**Comment by Tieqiong** [comment]
Just for reference, an easier way might be instead of installing gsl globally we can install it inside of the Conda environment by running `conda install gsl`. gsl is also one of the requirement for building for diffpy.pdffit2.

**Comment by sbillinge** [comment]
> Just for reference, an easier way might be instead of installing gsl globally we can install it inside of the Conda environment by running `conda install gsl`. gsl is also one of the requirement for building for diffpy.pdffit2.

yes, even better if it works.

**Comment by bobleesj** [comment]
doesn't seem to be relevant for diffpy.pdfgui since `diffpy.pdffit2` will be installed via conda-forge in the conda environment using `conda install --file requirements/run.txt`

**Comment by sbillinge** [comment]
we may want to just summarize the thread and dump it onto diffpy-users for safe keeping.  I htink that is a better place for it as it will affect only a small number of people.

**Comment by sbillinge** [comment]
closed by #211
